addons/pos_self_order/static/src/common/self_order_service.js

```diff
@@ -61,6 +61,24 @@ export class selfOrderCommon extends Reactive {
         }, {});
     }
 
+    handleProductChanges(payload) {
+        const product = new Product(payload.product, this.show_prices_with_tax_included);
+        this.productByIds[payload.product.id] = product;
+        for (const categ_name of payload.product.pos_categ_ids) {
+            if (!this.pos_category.map((c) => c.name).includes(categ_name)) {
+                continue;
+            }
+            const index = this.productsGroupedByCategory[categ_name].findIndex(
+                (p) => p.id === product.id
+            );
+            if (index >= 0) {
+                this.productsGroupedByCategory[categ_name][index] = product;
+            } else {
+                this.productsGroupedByCategory[categ_name].push(product);
+            }
+        }
```

Comment:
Assuming you only send your status: 

```suggestion
        const productId = payload.product_id;
        const newState = payload.status;
        const product = this.productByIds[productId];
        product.self_order_available = newState;
```